core/src/main/java/com/novoda/merlin/service/MerlinService.java

```diff
@@ -31,6 +31,9 @@ public IBinder onBind(Intent intent) {
     public boolean onUnbind(Intent intent) {
         isBound = false;
         connectivityChangesRegister.unregister();
+        connectivityChangesForwarder = null;
```

Comment:
Ah, this is an interesting point. 

The `MerlinServiceBinder` controls the lifecycle of this `MerlinService` so that these dependencies are not null. The `start` is called from the `LocalBinder` which is called by the `MerlinServiceBinder` which occurs when the `MerlinService` is bound. 

I guess these dependencies could feasibly be `null` if the `MerlinService` is bound and then it is `unbound` before the `MerlinServiceBinder` can `start` the work on the `MerlinService`. 

Is this likely to occur?  
I've made the dependency a bit more obvious by adding a check to the `LocalBinder` to ensure that these dependencies are bound.